### ISSUE TITLE
Make module configurations loosely coupled

### ISSUE THREAD

**Issue by markdumay** [gethinode/hinode#948]
## Problem or enhancement idea

Hinode supports various [modules](https://gethinode.com/docs/advanced-settings/module-development/) to enhance its functionality. These modules include core functionality, such as core Bootstrap ](https://getbootstrap.com/) , [FlexSearch ](https://github.com/nextapps-de/flexsearch), and [Font Awesome](https://fontawesome.com/). Other modules such as [Leaflet](https://github.com/gethinode/mod-leaflet) and [KaTeX](https://github.com/gethinode/mod-katex) can be included on a page-by-page basis.

The [module configuration ](https://gethinode.com/docs/configuration/modules/#configuring-modules)is defined in the site's parameters. It consists of a mix of general parameters and module-specific parameters. The current configuration is not fully modular. For example, the `disableTemplate` setting is an array of one or more values that need to be refined for each module that is added or removed from the module configuration.

## Proposed solution

The module configuration should be shifted to the configuration of the individual modules. You would only have to modify `[[module.imports]]`, as the reference module would include sensible default values. You could still override these default settings in the site's settings.

For example, the current module configuration of `KaTeX` is the following:

```toml
[modules]
    core = ["bootstrap", "flexsearch", "fontawesome"]
    optional = ["leaflet", "katex", "lottie"]
    excludeSCSS = ["bootstrap"]
    disableTemplate = ["katex"]
    [modules.katex]
        state = "defer"
```

The values `core`, `optional`, `excludeSCSS`, and `disableTemplate` would be shifted to the individual modules. The new configuration would be the following:

```toml
[modules]
    [modules.katex]
        integration = 'optional' // 'core' | 'critical' | 'optional'
        disableTemplate = true
        state = "defer"
    [...]
```

## Alternatives

N/A

## Additional context

See https://github.com/gethinode/hinode/discussions/944#discussioncomment-9532406 for more background.

**Comment by peters-r** [comment]
👍

**Comment by markdumay** [comment]
@peters-r the feature has been released in [v0.24.0-beta2](https://github.com/gethinode/hinode/releases/tag/v0.24.0-beta2). I'm doing some final integration tests, but so far, so good.

**Comment by markdumay** [comment]
Released in [v0.24.0](https://github.com/gethinode/hinode/releases/tag/v0.24.0)
